src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -497,7 +493,7 @@ void SpatialPooler::compute(const UInt inputArray[], bool learn, UInt activeArra
 void SpatialPooler::compute(SDR &input, bool learn, SDR &active) {
   updateBookeepingVars_(learn);
   calculateOverlap_(input, overlaps_);
-  calculateOverlapPct_(overlaps_, overlapsPct_);
+  // calculateOverlapPct_(overlaps_, overlapsPct_);
```

Comment:
wow, nice spot!! 
What are your intended use-cases? Removal could get rid off a few more variables...

Reply:
Even though variable `overlapsPct_` is unused I would like to keep it around.  It's a form of boosting, and I've seen it used in papers/models which do not use any other form of  boosting, such as (Kropff & Treves 2008), so it would be useful for reproducing other ppl's work.  I also want to keep it for future experiments.

Reply:
maybe then make this method compute it and keep it updated (if this wasn't the performance bottleneck (?))

Reply:
This was definately a bottleneck.  The issue is that the connections class does not keep track of the # of connected synapses in each segment.  Its a trivial amount of book keeping.

I would rather not add the bookkeeping to connections in this PR because I hope in the future to rewrite connections to keep its data in a different format, which would replace any logic I add for this now.  This future change would be to keep connected and disconnected synapses in separate lists, for speed.

Reply:
> I would rather not add the bookkeeping to connections in this PR because I hope in the future to rewrite connections to keep its data in a different format, which would replace any logic I add for this now. This future change would be to keep connected and disconnected synapses in separate lists, for speed. 

Definitely good approach, and I'm  looking forward for the next improvements! :)

Reply:
Nevermind, I will add this the book keeping for this.  The optimization I'm thinking of does not effect the postsynapses, just the presynapses.

Reply:
See issue #161